src/org/openbravo/erpCommon/ad_forms/AcctServer.java

```diff
@@ -1662,6 +1663,10 @@ public void setC_Period_ID() {
           .getId();
       data = AcctServerData.selectPeriodOpen(connectionProvider, AD_Client_ID, DocumentType,
           strOrgCalendarOwner, DateAcct);
+      if(data.length == 0) {
+        C_Period_ID = "";
```

Comment:
Are you sure that you must set this variable as an empty string instead a null value?

Reply:
After this method's execution, there's another method that checks if the variable is specifically ""